[DPE-6706] - Add Client Relation docs (#62)
canonical/charmed-etcd-operator: docs/how-to/client-relations.md

        )
```

Finally, define a callback function to handle the `authentication_updated` event. This function will be called when the relation is established and the authentication information is available. You can use this information to connect to etcd and perform any necessary operations.

```python
class MyCharm(CharmBase):
    def __init__(self, *args):
        super().__init__(*args)
        self.etcd = EtcdRequires(
            self,
            relation_name="etcd",
            prefix="/my-charm/",
            mtls_chain=self.model.config["mtls_chain"],
        )
        self.etcd.on.authentication_updated.connect(self._on_authentication_updated)

    def _on_authentication_updated(self, event):
        pass
```

The `authentication_updated` event provides the following attributes:

- `username`: The username of the user created in etcd. This matches the common name of the client certificate.
- `tls_ca`: The CA certificate used to sign the server certificate.

The `EtcdRequires` class also emit:

- `endpoints_changed` event when the endpoints of etcd change.
- `etcd_version_updated` event when the version of etcd changes.

> To integrate your charm with etcd, you need to establish a relation between your charm and the etcd charm. This can be done using the `juju integrate` command:

Reviewer: ```suggestion
To establish a relation between your charm and etcd, use the `juju integrate` command:
```
We can be more concise/straightforward in situations like this, especially since this is not a tutorial.

```{terminal}
:input: juju integrate charmed-etcd <your-charm>
```

To remove the relation, you can use the `juju remove-relation` command:

```{terminal}
:input: juju remove-relation <your-charm> charmed-etcd
```

If at any point in the charm code you need to access any field you can do the following:

```python
credentials = {
    "prefix": self.etcd.fetch_my_relation_field(relation.id, "prefix"),
    "endpoints": self.etcd.fetch_relation_field(self.etcd_relation.id, "endpoints"),
    "username": self.etcd.fetch_relation_field(self.etcd_relation.id, "username"),
    "tls-ca": self.etcd.fetch_relation_field(self.etcd_relation.id, "tls-ca"),
    "version": self.etcd.fetch_relation_field(self.etcd_relation.id, "version"),
}
```

## Integrate an application outside of Juju with etcd using data-integrator

> The [`data-integrator`](https://charmhub.io/data-integrator) charm is a bare-bones charm that allows for central management of database users, providing support for different kinds of data platform products (e.g. MongoDB, MySQL, PostgreSQL, Kafka, etc) with a consistent, opinionated and robust user experience.

Reviewer: ```suggestion
[`data-integrator`](https://charmhub.io/data-integrator) is a bare-bones charm that allows for central management of database users.
```
The reader can get more information from the link you provided.
